lighthouse-core/gather/driver.js

```diff
@@ -59,6 +59,10 @@ const DEFAULT_PROTOCOL_TIMEOUT = 30000;
  */
 
 class Driver {
+  online = true;
+  // eslint-disable-next-line no-invalid-this
+  fetcher = new Fetcher(this);
```

Comment:
I think this is an error in babel-eslint parser. `this` is perfectly fine here.

Reply:
> I think this is an error in babel-eslint parser. `this` is perfectly fine here.

even if it's parsed correctly, I'd imagine eslint isn't going to have any "allow in field initializers" code until they add support for public fields.

(there's also a corner case to handle, where `this` works here unless the constructor ends up returning something else. hopefully we never land code like that in lighthouse, though :)

Reply:
what do you mean a constructor can return something else?!

Reply:
wtf gross https://javascript.info/constructor-new#return-from-constructors

Reply:
> wtf gross

haha yes, maybe there's some purity argument somewhere since every function is a potential constructor, but it's a bad argument :)